toolchain/check/testdata/interop/cpp/class/method.carbon

```diff
@@ -67,10 +67,12 @@ fn Value(v: Cpp.HasQualifiers) {
   // CHECK:STDERR:
   v.volatile_this();
 
-  // CHECK:STDERR: fail_bad_object_param_qualifiers_by_value.carbon:[[@LINE+5]]:3: error: `addr self` method cannot be invoked on a value [AddrSelfIsNonRef]
+  // CHECK:STDERR: fail_bad_object_param_qualifiers_by_value.carbon:[[@LINE+7]]:3: error: no matching function for call to `ref_this` [CppOverloadingNoViableFunctionFound]
```

Comment:
Is this change expected? It seems like a regression in the diagnostic quality, and it's a little surprising that it doesn't affect `v.plain()`.

Reply:
Yes -- this is now failing in overload resolution rather than when forming the call on the Carbon side. We have a TODO in cpp_overload_resolution.cpp to improve the diagnostic here to list the candidates and explain why they're non-viable.